File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length()!=t.length()) return false;

        int[] charCount = new int[26];

        for(int i=0;i<s.length();i++){
            charCount[(int)s.charAt(i)-'a']++;
            charCount[(int)t.charAt(i)-'a']--;
        }

        for(int i: charCount){
            if(i!=0) return false;
        }

        return true;

    }
}

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False
        map = dict()
        for c in s:
            if c not in map:
                map[c] = 0
            map[c] += 1
        for c in t:
            if c in map:
                map[c] -= 1
                if map[c] == 0:
                    del map[c]
        return len(map)==0